misc/formatter/modules/formatter-cli/src/main/java/org/ballerinalang/formatter/cli/FormatUtil.java

```diff
@@ -0,0 +1,358 @@
+        }
+    }
+
+    private static void generateChangeReport(List<String> formattedFiles, boolean dryRun) {
+        if (!formattedFiles.isEmpty()) {
+            StringBuilder fileList = new StringBuilder();
+            if (dryRun) {
+                fileList.append(Messages.getFilesToModify()).append(System.lineSeparator());
+            } else {
+                fileList.append(Messages.getModifiedFiles()).append(System.lineSeparator());
+            }
+            for (String file : formattedFiles) {
+                fileList.append(file).append(System.lineSeparator());
+            }
+            outStream.println(fileList.toString());
+            if (!dryRun) {
+                outStream.println(Messages.getSuccessMessage());
+            }
+        } else {
+            outStream.println(Messages.getNoChanges());
+        }
+    }
+
+    private static String getModuleName(String moduleName) {
+        String pattern = Pattern.quote(File.separator);
+        String[] splitedTokens = moduleName.split(pattern);
+        return splitedTokens[splitedTokens.length - 1];
+    }
+
+    /**
+     * Compile whole ballerina project.
+     *
+     * @param sourceRoot source root
+     * @return {@link List<BLangPackage>} list of BLangPackages
+     */
+    private static List<BLangPackage> compileProject(Path sourceRoot) throws UnsupportedEncodingException {
+        emptyPrintStream = new EmptyPrintStream();
+        CompilerContext context = getCompilerContext(sourceRoot);
+        Compiler compiler = Compiler.getInstance(context);
+        // Set an EmptyPrintStream to hide unnecessary outputs from compiler.
+        compiler.setOutStream(emptyPrintStream);
+        return compiler.compilePackages(false);
+    }
+
+    /**
+     * Compile only a ballerina module.
+     *
+     * @param sourceRoot source root
+     * @param moduleName name of the module to be compiled
+     * @return {@link BLangPackage} ballerina package
+     */
+    private static BLangPackage compileModule(Path sourceRoot, String moduleName) throws UnsupportedEncodingException {
+        emptyPrintStream = new EmptyPrintStream();
+        CompilerContext context = getCompilerContext(sourceRoot);
+        Compiler compiler = Compiler.getInstance(context);
+        // Set an EmptyPrintStream to hide unnecessary outputs from compiler.
+        compiler.setOutStream(emptyPrintStream);
+        return compiler.compile(moduleName);
+    }
+
+    private static void formatAndWrite(BLangCompilationUnit compilationUnit, Path sourceRootPath,
+                                       List<String> formattedFiles, boolean dryRun) throws IOException {
+        String fileName = sourceRootPath.toString() + File.separator
+                + "src"
+                + File.separator
+                + compilationUnit.getPosition().getSource().getPackageName()
+                + File.separator
+                + compilationUnit.getPosition().getSource().getCompilationUnitName();
```

Comment:
If there a bal files inside nested directories of the module directory, do they get visited ?

Reply:
This `formatAndWrite` function is called iteratively for each `BLangCompilationUnit`. The `getCompilationUnits()` function used to obtain the compilation units resolves the `.bal` files residing inside subdirectories as well.